core/variant/variant_parser.cpp

```diff
@@ -2165,9 +2170,20 @@ Error VariantWriter::write(const Variant &p_variant, StoreStringFunc p_store_str
 					} else {
 						p_store_string_func(p_store_string_ud, ",");
 					}
-
+					Variant v = obj->get(E.name);
+					if (v.get_type() == Variant::OBJECT) {
+						if (v.get_validated_object() == obj) {
+							p_store_string_func(p_store_string_ud, "\"" + String("self"));
+							continue;
+						}
+						if (used_objects.has(obj)) {
+							p_store_string_func(p_store_string_ud, "\"" + obj->get_class() + "\": Recursive object");
```

Comment:
```suggestion
							p_store_string_func(p_store_string_ud, "\"" + E.name + "\":Recursive object");
```
To keep original property name instead of using class name